torch/csrc/utils/tensor_new.cpp

```diff
@@ -625,13 +618,13 @@ Tensor sparse_csr_tensor_ctor(c10::DispatchKey dispatch_key, at::ScalarType scal
                                            r.pyobject(VALUES_ARG), /*copy_variables=*/false, /*copy_numpy=*/true,
                                            /*type_inference=*/type_inference);
     Tensor crow_indices =  internal_new_from_data(values.options(),
-      crow_indices_scalar_type, r.deviceOptional(DEVICE_TYPE_ARG), r.pyobject(CROW_INDICES_ARG),
+      kInt, r.deviceOptional(DEVICE_TYPE_ARG), r.pyobject(CROW_INDICES_ARG),
```

Comment:
Does this mean that `crow_indices` and `col_indices` will be forced to have `int32` dtype? If so, it is a regression,

Reply:
Actually for `crow_indices` and `col_indices` the data type will be inferred  based on the data type of the input (tensor, list or numpy array). In previous code there was a bug processing dtype, moreover  the `dtype` was forced to be int32 specifically for `list` inputs.